Core/src/IfcGeometryConverter/SplineConverter.h

```diff
@@ -143,7 +144,110 @@ namespace OpenInfraPlatform
 						throw oip::UnhandledException(splineSurface);
 					}
 
+					/*! \brief This function calls other functions which aren't part of the current program flow.
+					 *
+					 * At the moment, \c computeCurvatureOfIfcBSplineCurveWithKnots is called from there.
+					 * \note This function will be removed if the called functions are part of the usual program flow.
+					 *
+					 * param[in] bspline A \c computeCurvatureOfIfcBSplineCurveWithKnots entity
+					 */
+					void junkfunctionToCallForCompiler(const EXPRESSReference<typename IfcEntityTypesT::IfcBSplineCurveWithKnots>& bspline) const throw(...)
+					{
+						const int order = bspline->Degree + 1;
+						const int numControlPoints = bspline->ControlPointsList.size();
+						const int numKnotsArray = order + numControlPoints;
+						const std::vector<double> knotArray = loadKnotArray(bspline, numKnotsArray);
+
+						const std::vector<carve::geom::vector<3>> controlPoints = loadControlPoints(bspline);
+
+						std::vector<double> length;
+						std::vector<double> curvature;
+						std::tie(length, curvature) = computeCurvatureOfBSplineCurveWithKnots(order, controlPoints, knotArray);
+					}
+
+					/*! \brief Computes the length and curvature of an B-Spline curve
+					 *
+					 * This function calculates the curvature of a B-Spline; rational B-Splines can't be handled from this function. 
+					 * As well, a vector with the corresponding curve length is calculated. 
+					 * The \c length is necessary to display or analyse the \c curvature in dependency to the true curve length.
+					 * The parameter \c t along a B-Spline curve doesn't represent the true length along the curve because the 
+					 * calculated curve points aren't evenly spaced.\n
+					 * Both return vectors have the same size. They come back as a \c std::tuple in the order \c length, \c curvature.
+					 *
+					 * \param[in]	order			Order of the B-Spline or rather the basis functions ( =degree+1 )
+					 * \param[in]	controlPoints	The vector of the B-Spline control points.
+					 * \param[in]	knotArray		The array / vector of knots.
+					 *
+					 * \return		Vector with curve length
+					 * \return		Vector with curve curvature
+					 */
+					std::tuple<std::vector<double>, std::vector<double>> computeCurvatureOfBSplineCurveWithKnots(

```

Comment:
My suggestion would be to rather use `std::vector<std::pair<double, double>>` if you're having pairs with "length along the curve" and its corresponding "curvature at that point". Access the member by `.first` and `.second` .

Reply:
Changed in commit 637e6d8